### PR TITLE
Depth axis

### LINKED ISSUE #40: depth axis scale/reading

### PR DESCRIPTION
**Pull request by jaxxzer**
fix #40

### CODE REVIEW

File: qml/DepthAxis.qml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }
+
+    Image {
+        anchors.right: parent.right
+        source: "/icons/depth_indicator.svg"
+        height: 20
+        width: 20
+        y: (parent.height / length_mm) * depth_mm - height / 2
+    }
```
Comment (patrickelectric) [inline qml/DepthAxis.qml]: For simple things like triangle, or square, you can use canvas:
```cpp
    Canvas {
        id: depthPointer
        y: (parent.height / length_mm) * depth_mm - height / 2
        z: root.z + 1
        width: 20
        height: width
        onWidthChanged: requestPaint()
        onPaint: {
            print(width, height)
            var ctx = getContext("2d")
            ctx.fillStyle = "red"
            ctx.beginPath()
            ctx.moveTo(width, 0)
            ctx.lineTo(0, height/2)
            ctx.lineTo(width, height)
            ctx.fill()
        }
    }
```

File: qml/DepthAxis.qml [hunk 2]
```diff
@@ -0,0 +1,128 @@
+import QtQuick 2.7
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.3
+
+import Logger 1.0
```
Comment (patrickelectric) [inline qml/DepthAxis.qml]: Why logger is necessary here ?

File: qml/DepthAxis.qml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            height: lastIndex ? remainingHeight() : nominalHeight()
+            Layout.fillHeight: true
+            Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+
+            property var lastIndex: (index + 1) == numTicks
+
+            Text {
+                id: label
+                //anchors.fill:parent // this text object is not anchored
+                //Layout.fillHeight: true
```
Comment (patrickelectric) [inline qml/DepthAxis.qml]: This code is for debug or can be deleted ?

Comment (jaxxzer) [reply qml/DepthAxis.qml]: It is a reminder for anyone who adjusts it in the future. It is a little bit nonconventional how I got it to look right I think.

Comment (jaxxzer) [reply qml/DepthAxis.qml]: I can try to rework it

File: qml/DepthAxis.qml (hunk 2 above)
Comment (jaxxzer) [reply qml/DepthAxis.qml]: copy paste

File: qml/DepthAxis.qml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    property var length_mm: end_mm - start_mm
+    property var depth_mm: 0
+
+    property var validIncrements: [0.1, 0.25, 0.5, 1, 2, 5, 10]
+
+    property int numTicks: 10
+    property int maxTicks: 10
+    property var increment: 1
+
+    property var color: "white"
```
[the remark this answers is not in the document]
Comment (jaxxzer) [reply qml/DepthAxis.qml]: where is Style.color determined?

File: qml/DepthAxis.qml (hunk 1 above)
Comment (jaxxzer) [reply qml/DepthAxis.qml]: That would require thought and effort

Comment (patrickelectric) [reply qml/DepthAxis.qml]: This code already do the red triangle. You just need to copy and paste :)

File: qml/DepthAxis.qml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jaxxzer) [reply qml/DepthAxis.qml]: The parent can set the color property of this item to Style.color if necessary. Style.color is giving me black which is not very good to see for most themes.

File: qml/DepthAxis.qml (hunk 1 above)
Comment (jaxxzer) [reply qml/DepthAxis.qml]: thanks for doing the hard work for me

File: qml/DepthAxis.qml (hunk 3 above)
Comment (jaxxzer) [reply qml/DepthAxis.qml]: There are some bugs with what I expect to be the reasonable thing to do (bind text anchors to parent, use horizontal and verticalAlignment). It hangs the program when trying to use these. I left the comments just to explain the situation.

**Comment by jaxxzer** [comment]
It's fixed. I think we should try some real life tests with this version and change it subsequently if we decide it's not right.

**Comment by jaxxzer** [comment]
@patrickelectric delete branches after merge please.

**Pull request merged**